### ISSUE TITLE
Schalter {$IFNDEF ZUGFeRD_Support} in "LoadFromXMLDocument"

### ISSUE THREAD

**Issue by dsshad** [LandrixSoftware/XRechnung-for-Delphi#12]
Hallo,
müsste der Schalter {$IF**N**DEF ZUGFeRD_Support} in "class function TXRechnungInvoiceAdapter.LoadFromXMLDocument" in "intf.XRechnung.pas" nicht eigentlich {$IFDEF ZUGFeRD_Support} lauten? Also umgekehrt ohne N?

**Comment by landrix** [comment]
Nein, warum? Ohne ZUGFeRD Support wird der eigene Adapter TXRechnungInvoiceAdapter verwendet, ansonsten der für die ZUGFeRD-Lib.

**Comment by dsshad** [comment]
Ok, verstehe.

Der Grund für meine Frage war, dass beim Versuch eine XRechnung mit diesen Optionen zu erstellen

- Ausgabeformat: ZUGFeRD
- Ausgabeversion: 3.0.1
- Zahlungsbedingungen: 1 Skonto
- Nachlaesse/Zuschlaege: ja

beim anschließenden Vergleich des Originals mit der Testrechnung Folgendes "bemängelt" wurde:

original
![original](https://github.com/user-attachments/assets/9e3a0bef-68f9-4dd9-af6a-816ffd43d98f)

Testrechnung
![testrechnung](https://github.com/user-attachments/assets/e90e16c4-33a1-4c7b-ba08-8ac76040050a)

Erst dachte ich, dass es an diesem Abschnitt in "class function TZUGFeRDInvoiceAdapter.LoadFromInvoiceDescriptor" liegt:
  //TODO #SKONTO Type
  ...
  ...
+ die Stelle, wo dann im Falle von iptt_Net '#' durch ' ' ersetzt wird.

und dass es ausgehend vom Kommentar einfach nur generell ein noch offener Bereich ist.

Dann erschien mir aber auf den ersten Blick die andere Stelle als vertauscht. Hatte es ohne N mal probeweise gemacht und da stimmte die Testrechnung mit dem Original überein, weil dadurch u.a. "class procedure TXRechnungInvoiceAdapter301.InternalReadPaymentTerms" aufgerufen wurde, wo die Skonto-Informationen verarbeitet wurden.

**Comment by landrix** [comment]
Das liegt daran, dass die Angabe von Zahlungsdaten in XRechnung anders gelöst ist als in ZUGFeRD generell, bzw es gibt zwei Möglichkeiten der Abbildung. Ich werde versuchen, dass in der ZUGFeRD-Lib nachzubilden. In der originalen ZUGFeRD-C#-Lib wird darauf nicht eingegangen.

**Comment by landrix** [comment]
ok, muss ich nochmal prüfen

**Comment by dsshad** [comment]
Wenn das Ziel der letzten Änderungen zum Thema "Skonto" war, das Problem aus diesem Eintrag zu lösen, dann hat es funktioniert, so wie aussieht.

Habe den Fall nochmal nachgestellt und nun kommt keine Meldung mehr beim Erstellen einer Ausgangsrechnung im ZUGFeRD-Format, dass Original- und Testrechnung nicht identisch sind. Wenn man diese Ausgangsrechnung dann wie eine Eingangsrechnung einliest, sind jetzt auch die Skontodetails da (vorher fehleten sie).
